airflow/www/app.py

```diff
@@ -95,9 +96,18 @@ def create_app(config=None, session=None, testing=False, app_name="Airflow"):
                 """Your CUSTOM_SECURITY_MANAGER must now extend AirflowSecurityManager,
                  not FAB's security manager.""")
 
-        appbuilder = AppBuilder(
-            app,
-            db.session if not session else session,
+        class AirflowAppBuilder(AppBuilder):
+
+            def add_view(self, baseview, *args, **kwargs):
+                if hasattr(baseview, 'datamodel'):
+                    # Delete sessions if initiated previously to limit side effects. We want to use
+                    # the current session in the current application.
+                    baseview.datamodel.session = None
```

Comment:
The session is saved to the object, which is an attribute of the class (not instance), so we have a huge side effect.
https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/base.py#L339-L347
https://github.com/apache/airflow/blob/df159826d61a934862e9fe6a62b535eb443e5710/airflow/www/views.py#L2309

Reply:
I think I had to work around this (or at least it bit me) when I was doing the TZ work in the UI

:+1:

Reply:
Oh, i remember what my problem was: date columns on FABs own security models.